### PR TITLE
feat: validity check page

### PR DESCRIPTION
**Pull request by sebastianquek**
## Summary
This PR accomplishes the following:
* Add `getStatusProps` as a util for `Validity` related components. This allows us to make use of shared props for each `CheckStatus`: https://gist.github.com/sebastianquek/966e1bef9306420eef36fb49d1c0f946
* Implement validity check page:
  * After scanning using `QrScannerScreen`, the user will be routed to `ValidityCheckScreen`.
  * If document is valid, then he will be routed to `ScannedDocumentScreen`, if not, he remains on `ValidityCheckScreen` 
* Saving a document will not trigger the same verification checks again
* Create `ValidityCard` which is used by `ValidityCheckScreen`
* Add `hasBorder` prop to `Header`

### Folder structure of `Validity`
With the new `ValidityCard`, I've moved several of the validity related files to a `Validity` folder. For example, previously we had a `constants/verifier.ts` file which contained the `CheckStatus` enum. I've moved this to `Validity` instead.

```
src/components/Validity
├── ValidityBanner
│   ├── ValidityBanner.test.tsx
│   ├── ValidityBanner.tsx
│   ├── ValidityBannerCheckItem.test.tsx
│   ├── ValidityBannerCheckItem.tsx
│   ├── ValidityBannerContent.tsx
│   ├── ValidityBannerHeader.test.tsx
│   ├── ValidityBannerHeader.tsx
│   └── index.tsx
├── ValidityCard
│   ├── ValidityCard.test.tsx
│   ├── ValidityCard.tsx
│   ├── ValidityCardCheckItem.tsx
│   ├── ValidityCardHeader.tsx
│   └── index.tsx
├── ValidityIcon.tsx
├── constants.ts
├── index.tsx
└── utils.ts
```
- [x] Intermediate step after scanning QR code, but before rendering the document
- [x] Use different validity check styling
- [x] Update header styling (remove underline)

### CODE REVIEW

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=32

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-32

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=32

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-32

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=32

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-32

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=32

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-32

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=32

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-32

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=32

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-32

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=32

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-32

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=32

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-32

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=32

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-32

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=32

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-32

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=32

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-32

File: src/components/DocumentRenderer/ScannedDocumentRendererContainer.tsx [hunk 1]
```diff
@@ -6,16 +6,15 @@ import { SignedDocument, getData } from "@govtechsg/open-attestation";
...
+import { CheckStatus } from "../Validity";
 
 export const ScannedDocumentRendererContainer: FunctionComponent<NavigationProps> = ({
   navigation
 }) => {
   const { db } = useDbContext();
   const document: SignedDocument = navigation.getParam("document");
   const isSavable: boolean = navigation.getParam("savable");
-  const verificationStatuses = useDocumentVerifier(document as SignedDocument);
+  const verificationStatuses = navigation.getParam("verificationStatuses");
```
Comment (yehjxraymond) [inline src/components/DocumentRenderer/ScannedDocumentRendererContainer.tsx]: Would you like to type this?

File: src/components/ValidityCheckScreen/ValidityCheckScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import React, { FunctionComponent, useEffect } from "react";
```
Comment (yehjxraymond) [inline src/components/ValidityCheckScreen/ValidityCheckScreen.tsx]: Just wondering about the folder structure, seemed weird that this is the only one `Screen` component in its own folder? Probably can look into the folder struc in a separate PR.

Comment (sebastianquek) [reply src/components/ValidityCheckScreen/ValidityCheckScreen.tsx]: I'll move it into `QrScanner`, we can revisit this again in future

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=32

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-32

**Pull request merged**